From 3bfec15358b788a2b7e1beefaa177ba84a6da328 Mon Sep 17 00:00:00 2001
From: Ted Meyer <tmathmeyer@chromium.org>
Date: Mon, 1 Jun 2020 19:31:18 +0000
Subject: [PATCH] Remove NotifyError calls and just send a normal message
 instead

rewrote the TODO as well.

Bug: 1088055
Change-Id: If83ceeb3a786a1813e41bd3ac6c7309791e92400
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2223010
Auto-Submit: Ted Meyer <tmathmeyer@chromium.org>
Commit-Queue: Dale Curtis <dalecurtis@chromium.org>
Reviewed-by: Dale Curtis <dalecurtis@chromium.org>
Cr-Commit-Position: refs/heads/master@{#773776}
---
 media/filters/decoder_selector.cc | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/media/filters/decoder_selector.cc b/media/filters/decoder_selector.cc
index bb06378b92d..9f510bb578c 100644
--- media/filters/decoder_selector.cc
+++ media/filters/decoder_selector.cc
@@ -165,13 +165,11 @@ void DecoderSelector<StreamType>::OnDecoderInitializeDone(Status status) {
   DCHECK(task_runner_->BelongsToCurrentThread());
 
   if (!status.is_ok()) {
-    // TODO(tmathmeyer) this might be noisy in media log. Consider batching
-    // all failures as causes to a single Status object and only surfacing it if
-    // decoder selection fails entirely.
-    media_log_->NotifyError(
-        Status(StatusCode::kDecoderFailedInitialization)
-            .WithData("Decoder name", decoder_->GetDisplayName())
-            .AddCause(std::move(status)));
+    // TODO(tmathmeyer) this was too noisy in media log. Batch all the logs
+    // together and then send them as an informational notice instead of
+    // using NotifyError.
+    MEDIA_LOG(INFO, media_log_)
+        << "Failed to initialize " << decoder_->GetDisplayName();
 
     // Try the next decoder on the list.
     decoder_.reset();
